modin/pandas/groupby.py

```diff
@@ -682,14 +682,34 @@ def dtypes(self):
             )
         )
 
-    def first(self, numeric_only=False, min_count=-1):
+    def first(self, numeric_only=False, min_count=-1, skipna=True):
+        # TODO: check if it works properly if we simply pass 'skipna' to
+        # the query compiler level
+        if not skipna:
+            return self._default_to_pandas(
```

Comment:
For the `_default_to_pandas` cases, can we put that behavior in the query compiler?

In general, we are trying to avoid defaulting to pandas in the API layer because some implementations in the future may not want to default to pandas.

This will require adding a `BaseQueryCompiler.groupby_first` API (and similar for the others)

Reply:
removed the call to d2p completely since it functions correctly when `skipna` is included as an argument in `agg_kwarg`

Reply:
Just a reminder, you didn't push anything yet 😄